doc/hardware/porting/board_porting.rst

```diff
@@ -523,90 +688,37 @@ Multiple board revisions
 See :ref:`application_board_version` for basics on this feature from the user
 perspective.
 
-To create a new board revision for the ``plank`` board, create these additional
-files in the board folder:
-
-.. code-block:: none
-
-   boards/<ARCH>/plank
-   ├── plank_<revision>.conf     # optional
-   ├── plank_<revision>.overlay  # optional
-   └── revision.cmake
-
-When the user builds for board ``plank@<revision>``:
-
-- The optional Kconfig settings specified in the file
-  :file:`plank_<revision>.conf` will be merged into the board's default Kconfig
-  configuration.
+Board revisions are described in the ``revision`` entry of the
+:file:`board.yml`.
 
-- The optional devicetree overlay :file:`plank_<revision>.overlay` will be added
-  to the common :file:`plank.dts` devicetree file
+.. code-block:: yaml
 
-- The :file:`revision.cmake` file controls how the Zephyr build system matches
-  the ``<board>@<revision>`` string specified by the user when building an
-  application for the board.
+   board:
+     revision:
+       format: <major.minor.patch|letter|number|custom>
+       default: <default-revision-value>
+       exact: <true|false>
+       revisions:
+       - name: <revA>
+       - name: <revB>
 
-Currently, ``<revision>`` can be either a numeric ``MAJOR.MINOR.PATCH`` style
-revision like ``1.5.0``, an integer number like ``1``, or single letter like
-``A``, ``B``, etc. Zephyr provides a CMake board extension function,
-``board_check_revision()``, to make it easy to match either style from
-:file:`revision.cmake`.
+Zephyr natively supports the following revision formats:
 
-Valid board revisions may be specified as arguments to the
-``board_check_revision()`` function, like:
-
-.. code-block:: cmake
-
-   board_check_revision(FORMAT MAJOR.MINOR.PATCH
-                        VALID_REVISIONS 0.1.0 0.3.0 ...
-   )
-
-.. note::
-   ``VALID_REVISIONS`` can be omitted if all valid revisions have specific
-   Kconfig fragments, such as ``<board>_0_1_0.conf``, ``<board>_0_3_0.conf``.
-   This allows you to just place Kconfig revision fragments in the board
-   folder and not have to keep the corresponding ``VALID_REVISIONS`` in sync.
-
-The following sections describe how to support these styles of revision
-numbers.
-
-MAJOR.MINOR.PATCH revisions
-===========================
-
-Let's say you want to add support for revisions ``0.5.0``, ``1.0.0``, and
-``1.5.0`` of the ``plank`` board with both Kconfig fragments and devicetree
-overlays. Create :file:`revision.cmake` with
-``board_check_revision(FORMAT MAJOR.MINOR.PATCH)``, and create the following
-additional files in the board directory:
-
-.. code-block:: none
+- ``major.minor.patch``: match a three digit revision, such as ``1.2.3``.
+- ``number``: match a single number.
+- ``letter``: match a single letter.
```

Comment:
should add a note that it only supports a single **uppercase** letter

Reply:
done